File: kings-api/model/Highlight.py

from model.BookEnum import BookEnum

class Highlight:
    def __init__(self, note: str, book: BookEnum, chapter: int, verse: str = None):
        self._note = note
        self._book = book
        self._chapter = chapter
        self._verse = verse

    def get_note(self) -> str:
        """Returns the note."""
        return self._note

    def get_book(self) -> BookEnum:
        """Returns the book."""
        return self._book

    def get_chapter(self) -> int:
        """Returns the chapter number."""
        return self._chapter

    def get_verse(self) -> str:
        """Returns the verse or verse range (if any)."""
        return self._verse

    def __str__(self):
        reference = f"{self._book.value} {self._chapter}"
        if self._verse:
            reference += f":{self._verse}"
        return f'Highlight: "{self._note}" - {reference}'

    def to_dict(self):
        """Converts highlight to a dictionary format"""
        return {
            "note": self._note,
            "book": self._book.value,
            "chapter": self._chapter,
            "verse": self._verse
        }

File: kings-api/src/main/java/bible/vision/kingsapi/model/Highlight.java
package bible.vision.kingsapi.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

@Entity
public class Highlight {
    @Id
    @GeneratedValue
    private String id;
    private BookEnum book;
    private String note;
    private int chapter;
    private String verse;

    public Highlight() {}
    public Highlight(BookEnum book, int chapter, String verse, String note) {
        this.book = book;
        this.chapter = chapter;
        this.verse = verse;
        this.note = note;
    }

    public String getId() {
        return id;
    }

    public BookEnum getBook() {
        return book;
    }

    public String getNote() {
        return note;
    }

    public int getChapter() {
        return chapter;
    }

    public String getVerse() {
        return verse;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setBook(BookEnum book) {
        this.book = book;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public void setChapter(int chapter) {
        this.chapter = chapter;
    }

    public void setVerse(String verse) {
        this.verse = verse;
    }

    @Override
    public String toString() {
        return "Highlight{" +
                note +
                '\'' + book + " " + chapter + ":" + verse +
                '}';
    }
}
